lib/travis/build/script/haskell.rb

```diff
@@ -3,19 +3,38 @@ module Build
     class Script
       class Haskell < Script
         DEFAULTS = {
-          ghc: Travis::Build.config.ghc_default.untaint
-        }
+          cabal: Travis::Build.config.cabal_default.to_s.untaint,
+          ghc: Travis::Build.config.ghc_default.to_s.untaint
+        }.freeze
+        GHC_VERSION_ALIASES = Travis::Build.config.ghc_version_aliases_hash.merge(
+          'default' => DEFAULTS[:ghc]
+        ).freeze
 
-        def setup
+        def configure
           super
           sh.raw(
             template(
               'haskell.sh',
               default_ghc: DEFAULTS[:ghc],
+              default_cabal: DEFAULTS[:cabal],
               root: '/'
             )
           )
-          sh.export 'PATH', "#{path}:$PATH", assert: true
+          # Automatic installation of exact versions *only*.
+          if version =~ /^\d+\.\d+\.\d+$/ && cabal_version =~ /^\d+\.\d+$/
```

Comment:
Are these numbers reasonable? Do we ever have version numbers in GHC and cabal that don't match these?